pyvista/core/dataset.py

```diff
@@ -1416,7 +1416,7 @@ def flip_z(
         """
         if point is None:
             point = self.center
-        check_valid_vector(point, 'point')
+        check_valid_vector(cast(VectorLike[float], point), 'point')
```

Comment:
@user27182 Is there any reason why this cast is needed? If needed, there are other places in this file where a cast is needed as well.

Reply:
I'll have to look into it. These changes were not directly related to this PR IIRC. If it's redundant this mypy option should fix this kind of thing

https://mypy.readthedocs.io/en/stable/config_file.html#confval-warn_redundant_casts

Reply:
The cast is not needed. Both casts added for this file were removed in https://github.com/pyvista/pyvista/pull/5571/commits/9420512fa39a38203a166f7f0944ebffb54db92e.